yarn-project/noir-protocol-circuits/src/crates/types/src/address.nr

```diff
@@ -164,19 +169,20 @@ impl PartialAddress {
         }
     }
 
-    pub fn compute(contract_address_salt : Field, function_tree_root : Field, constructor_hash : Field) -> Self {
+    pub fn compute(contract_class_id : Field, salt : Field, initialization_hash: Field, portal_contract_address: EthAddress) -> Self {
+        PartialAddress::compute_from_salted_initialization_hash(contract_class_id, SaltedInitializationHash::compute(salt, initialization_hash, portal_contract_address))
+    }
+
+    pub fn compute_from_salted_initialization_hash(contract_class_id : Field, salted_initialization_hash : SaltedInitializationHash) -> Self {
         PartialAddress::from_field(
             pedersen_hash([
-                // TODO why the zeroes?
-                0,
-                0,
-                contract_address_salt,
-                function_tree_root,
-                constructor_hash
+                contract_class_id,
+                salted_initialization_hash.to_field()
             ], GENERATOR_INDEX__PARTIAL_ADDRESS)
```

Comment:
Are these all the correct generator index values or does this need validating later on?

Reply:
Most definitely not the correct ones. I'm throwing in TODOs to review all generator indices.